policies/entity_association.rego

```diff
@@ -0,0 +1,26 @@
+package after_resolution
+import rego.v1
+
+# Collection of all entity names from the registry.
+known_entities := { g.name |
+    some g in input.groups
+    g.type == "resource"
+}
+
+# checks that all entity associations have valid groups.
+deny contains entity_association_violation(description, group.id) if {
+    group := input.groups[_]
+    some entity in group.entity_associations
+    not known_entities[entity]
+    description := sprintf("Unknown entity '%s' associated in group '%s'", [entity, group.id])
```

Comment:
we chatted about it on Wed tooling call and agreed weaver should check that the association should exist. 

Similar issue: https://github.com/open-telemetry/weaver/issues/673

Semconv can have additional policies (e.g. non-deprecated can only reference non-deprecated)

Reply:
Sounds good.  We'll leave this here and move it into weaver.